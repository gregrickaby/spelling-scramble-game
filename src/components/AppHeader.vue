<script setup lang="ts"></script>

<template>
  <header class="flex flex-col items-center justify-center gap-4">
    <h1>Spelling Scramble</h1>
  </header>
</template>
