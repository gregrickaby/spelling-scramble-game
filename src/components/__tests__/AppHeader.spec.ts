import AppHeader from '@/components/AppHeader.vue'
import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import type { RouteRecordRaw } from 'vue-router'
import { createRouter, createWebHistory } from 'vue-router'

/**
 * Define a route.
 */
const routes: Array<RouteRecordRaw> = [
  {
    path: '/',
    name: 'Home',
    component: {
      template: AppHeader
    }
  }
]

/**
 * Create a router instance.
 *
 * @see https://router.vuejs.org/api/#createrouter
 */
const router = createRouter({
  history: createWebHistory(),
  routes
})

/**
 * Define a test suite.
 *
 * @see https://vitest.dev/api/#describe
 */
describe('AppHeader', () => {
  /**
   * Mount the component.
   *
   * @see https://vue-test-utils.vuejs.org/api/#mount
   */
  const wrapper = mount(AppHeader, {
    global: {
      plugins: [router]
    }
  })

  /**
   * Assert the component renders.
   *
   * @see https://vitest.dev/api/expect.html
   */
  it('renders component', () => {
    expect(wrapper).toBeTruthy()
  })

  /**
   * Assert the component renders the title.
   */
  it('renders title', () => {
    expect(wrapper.text()).toContain('Spelling Scramble')
  })

  /**
   * Assert the component contains a link to the home page.
   */
  it('contains a link to the home page', async () => {
    await router.isReady()
    const link = wrapper.findComponent({ name: 'RouterLink' })
    expect(link.exists()).toBe(true)
    expect(link.props().to).toBe('/')
    expect(wrapper.find('a').attributes('href')).toBe('/')
  })

  /**
   * Assert the component matches the snapshot.
   *
   * @see https://vitest.dev/api/expect.html#tomatchsnapshot
   */
  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
