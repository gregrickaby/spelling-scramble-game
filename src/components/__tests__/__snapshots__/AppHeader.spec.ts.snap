// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppHeader > matches snapshot 1`] = `
"<header class="flex">
  <h1>Spelling Scramble</h1>
  <nav><a href="/" class="router-link-active router-link-exact-active" aria-current="page">Homes</a></nav>
</header>"
`;
